packages/gitbook/src/components/SitePage/SitePage.tsx

```diff
@@ -0,0 +1,191 @@
+    );
+    const withPageFeedback = customization.feedback.enabled;
+
+    const withSections = Boolean(sections && sections.list.length > 0);
+    const headerOffset = { sectionsHeader: withSections, topHeader: withTopHeader };
+
+    const document = await getPageDocument(context.dataFetcher, context.space.id, page);
+
+    return (
+        <>
+            {withFullPageCover && page.cover ? (
+                <PageCover as="full" page={page} cover={page.cover} context={context} />
+            ) : null}
+            {/* We use a flex row reverse to render the aside first because the page is streamed. */}
+            <div className="flex flex-row-reverse justify-end grow">
+                <PageAside
+                    page={page}
+                    document={document}
+                    withHeaderOffset={headerOffset}
+                    withFullPageCover={withFullPageCover}
+                    withPageFeedback={withPageFeedback}
+                    context={context}
+                />
+                <PageBody
+                    context={context}
+                    page={page}
+                    ancestors={ancestors}
+                    document={document}
+                    withPageFeedback={withPageFeedback}
+                />
+            </div>
+            <React.Suspense fallback={null}>
+                <PageClientLayout withSections={withSections} />
+            </React.Suspense>
+        </>
+    );
+}
+
+export async function generateSitePageViewport(context: GitBookSiteContext): Promise<Viewport> {
+    const { customization } = context;
+
+    return {
+        colorScheme: customization.themes.toggeable
+            ? customization.themes.default === CustomizationThemeMode.Dark
+                ? 'dark light'
+                : 'light dark'
+            : customization.themes.default,
+    };
+}
+
+export async function generateSitePageMetadata(props: SitePageProps): Promise<Metadata> {
+    const { context, pageTarget } = await getPageDataWithFallback({
+        context: props.context,
+        pagePathParams: props.pageParams,
+        fallback: props.fallback,
+    });
+
+    if (!pageTarget) {
+        notFound();
+    }
+
+    const { page, ancestors } = pageTarget;
+    const { site, customization, pages } = context;
+
+    return {
+        title: [page.title, site.title].filter(Boolean).join(' | '),
+        description: page.description ?? '',
+        alternates: {
+            // Trim trailing slashes in canonical URL to match the redirect behavior
+            canonical: (await getAbsoluteHref(getPagePath(pages, page), true)).replace(/\/+$/, ''),
+        },
+        openGraph: {
+            images: [
+                customization.socialPreview.url ??
+                    (await getAbsoluteHref(`~gitbook/ogimage/${page.id}`, true)),
+            ],
+        },
+        robots:
+            (await isSiteIndexable(site)) && isPageIndexable(ancestors, page)
```

Comment:
Lot of parallel opportunities here.

Reply:
It has no impact, async doesn't mean slow